model-optimizer/extensions/front/EmbeddingBagFuse.py

```diff
@@ -0,0 +1,91 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from extensions.ops.embedding_bag import EmbeddingBagOffsetsSum
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.graph.graph import Graph, rename_nodes
+
+
+class EmbeddingBagFuse(FrontReplacementSubgraph):
+    enabled = True
+
+    def run_after(self):
+        from extensions.front.ExpandDimsToUnsqueeze import ExpandDimsToUnsqueeze
+        from extensions.front.AttributedGatherNormalizer import AttributedGatherNormalizer
+        return [ExpandDimsToUnsqueeze, AttributedGatherNormalizer]
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('weights', dict(op='Const')),
+                ('concat_before', dict(op='Concat')),
+                ('gather_before1_1', dict(op='Gather')),
+                ('unsqueeze_before1_1', dict(op='Unsqueeze')),
+                ('gather_before2_1', dict(op='Gather')),
+                ('unsqueeze_before2_1', dict(op='Unsqueeze')),
+                ('slice1', dict(op='Slice')),
+                ('gather_after1', dict(op='Gather')),
+                ('reduce1', dict(op='ReduceSum')),
+                ('unsqueeze_after1', dict(op='Unsqueeze')),
+                ('concat_after', dict(op='Concat')),
+            ],
+            edges=[
+                ('concat_before', 'gather_before1_1'),
+                ('concat_before', 'gather_before2_1'),
+                ('gather_before1_1', 'unsqueeze_before1_1'),
+                ('gather_before2_1', 'unsqueeze_before2_1'),
+                ('unsqueeze_before1_1', 'slice1', {'out': 0, 'in': 1}),
+                ('unsqueeze_before2_1', 'slice1', {'out': 0, 'in': 2}),
+                ('weights', 'gather_after1', {'out': 0, 'in': 0}),
+                ('slice1', 'gather_after1', {'out': 0, 'in': 1}),
+                ('gather_after1', 'reduce1'),
+                ('reduce1', 'unsqueeze_after1'),
+                ('unsqueeze_after1', 'concat_after'),
+            ])
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        concat_before = match['concat_before']
+        gather_after1 = match['gather_after1']
+        slice1 = match['slice1']
+        concat_after = match['concat_after']
+        weights_node = gather_after1.in_port(0).get_source().node
+        gather_after_axis = gather_after1.in_port(2).get_source().node.soft_get('value')
+        for dst_port in weights_node.out_port(0).get_destinations():
+            node = dst_port.node
+            if node.op == 'Gather':
+                # validate that all Gathers have same axis
+                if node.in_port(2).get_source().node.soft_get('value') != gather_after_axis:
+                    return
+                dst_port.disconnect()
```

Comment:
you modify the graph even if you don't apply the transformation (returned from the previous line). Let's create a list of ports to disconnect (one list for lines 72 and 81) and disconnect all of them in line 82.